cmd/portforward.go

```diff
@@ -0,0 +1,548 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/mendersoftware/go-lib-micro/ws"
+	wsshell "github.com/mendersoftware/go-lib-micro/ws/shell"
+	"github.com/mendersoftware/mender-cli/client/deviceconnect"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/spf13/cobra"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	argBindHost    = "bind"
+	readBuffLength = 64 * 1024
+)
+
+const (
+	ProtoTypePortForward       = 3
+	MessageTypePortForwardNew  = "new"
+	MessageTypePortForwardStop = "stop"
+	MessageTypePortForward     = "forward"
+)
+
+const (
+	propertyConnectionID = "connection_id"
+	propertyRemoteHost   = "remote_host"
+	propertyRemotePort   = "remote_port"
+	propertyProtocol     = "protocol"
+)
+
+var portForwardCmd = &cobra.Command{
+	Use:   "port-forward DEVICE_ID LOCAL_PORT:REMOTE_PORT ...",
+	Short: "Forward one or more local ports to a device",
+	Args:  cobra.MinimumNArgs(2),
+	Run: func(c *cobra.Command, args []string) {
+		cmd, err := NewPortForwardCmd(c, args)
+		CheckErr(err)
+		CheckErr(cmd.Run())
+	},
+}
+
+func init() {
+	portForwardCmd.Flags().StringP(argBindHost, "", "127.0.0.1", "binding host")
```

Comment:
can we have 127.0.0.1 as a const/variable?

Reply:
done